src/script/printer/python_doc_printer.cc

```diff
@@ -31,6 +31,114 @@ namespace tvm {
 
+/*!
+ * \brief Operator precedence
+ *
+ * This is based on
+ * https://docs.python.org/3/reference/expressions.html#operator-precedence
+ */
+enum class ExprPrecedence : int32_t {
+  /*! \brief Unknown precedence */
+  kUnkown = 0,
+  /*! \brief Lambda Expression */
+  kLambda = 1,
+  /*! \brief Conditional Expression */
+  kIfThenElse = 2,
+  /*! \brief Boolean OR */
+  kBooleanOr = 3,
+  /*! \brief Boolean AND */
+  kBooleanAnd = 4,
+  /*! \brief Boolean NOT */
+  kBooleanNot = 5,
+  /*! \brief Comparisons */
+  kComparison = 6,
+  /*! \brief Bitwise OR */
+  kBitwiseOr = 7,
+  /*! \brief Bitwise XOR */
+  kBitwiseXor = 8,
+  /*! \brief Bitwise AND */
+  kBitwiseAnd = 9,
+  /*! \brief Shift Operators */
+  kShift = 10,
+  /*! \brief Addition and subtraction */
+  kAdd = 11,
+  /*! \brief Multiplication, division, floor division, remainder */
+  kMult = 12,
+  /*! \brief Positive negative and bitwise NOT */
+  kUnary = 13,
+  /*! \brief Exponentiation */
+  kExp = 14,
+  /*! \brief Index access, attribute access, call and atom expression */
+  kIdentity = 15,
+};
+
+#define DOC_PRECEDENCE_ENTRY(RefType, Precedence) \
+  { RefType::ContainerType::RuntimeTypeIndex(), ExprPrecedence::Precedence }
+
+ExprPrecedence GetExprPrecedence(const ExprDoc& doc) {
+  // Key is the value of OperationDocNode::Kind
+  static const std::vector<ExprPrecedence> op_kind_precedence = []() {
+    using OpKind = OperationDocNode::Kind;
+    std::map<OpKind, ExprPrecedence> raw_table = {
+        {OpKind::kUSub, ExprPrecedence::kUnary},             //
+        {OpKind::kInvert, ExprPrecedence::kUnary},           //
+        {OpKind::kAdd, ExprPrecedence::kAdd},                //
+        {OpKind::kSub, ExprPrecedence::kAdd},                //
+        {OpKind::kMult, ExprPrecedence::kMult},              //
+        {OpKind::kDiv, ExprPrecedence::kMult},               //
+        {OpKind::kFloorDiv, ExprPrecedence::kMult},          //
+        {OpKind::kMod, ExprPrecedence::kMult},               //
+        {OpKind::kPow, ExprPrecedence::kExp},                //
+        {OpKind::kLShift, ExprPrecedence::kShift},           //
+        {OpKind::kRShift, ExprPrecedence::kShift},           //
+        {OpKind::kBitAnd, ExprPrecedence::kBitwiseAnd},      //
+        {OpKind::kBitOr, ExprPrecedence::kBitwiseOr},        //
+        {OpKind::kBitXor, ExprPrecedence::kBitwiseXor},      //
+        {OpKind::kLt, ExprPrecedence::kComparison},          //
+        {OpKind::kLtE, ExprPrecedence::kComparison},         //
+        {OpKind::kEq, ExprPrecedence::kComparison},          //
+        {OpKind::kNotEq, ExprPrecedence::kComparison},       //
+        {OpKind::kGt, ExprPrecedence::kComparison},          //
+        {OpKind::kGtE, ExprPrecedence::kComparison},         //
+        {OpKind::kIfThenElse, ExprPrecedence::kIfThenElse},  //
+    };
+
+    std::vector<ExprPrecedence> table;
+    table.resize(static_cast<int>(OperationDocNode::Kind::kSpecialEnd) + 1);
```

Comment:
1. We may merge them into 1 stmt so `resize` is not needed;
2. Let's initialize every slot to "kUnknown"